tests/integration/kubernetes/confidential_kbs.sh

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2024 Red Hat
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Provides a library to deal with the CoCo KBS
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+kubernetes_dir="$(dirname "$(readlink -f "$0")")"
+# shellcheck disable=1091
+source "${kubernetes_dir}/../../gha-run-k8s-common.sh"
+
+# Where the kbs sources will be cloned
+readonly COCO_KBS_DIR="/tmp/kbs"
+
+# Delete the kbs on Kubernetes
+#
+# Note: assume the kbs sources were cloned to $COCO_KBS_DIR
+#
+function kbs_k8s_delete() {
+	pushd "$COCO_KBS_DIR"
+	kubectl delete -k kbs/config/kubernetes/overlays
+	popd
+}
+
+# Deploy the kbs on Kubernetes
+#
+function kbs_k8s_deploy() {
+	local image
+	local image_tag
+	local ns="coco-tenant"
+	local repo
+	local kbs_ip
+	local kbs_port
+	local version
+
+	# yq is needed by get_from_kata_deps
+	ensure_yq
+
+	# Read from versions.yaml
+	repo=$(get_from_kata_deps "externals.coco-kbs.url")
+	version=$(get_from_kata_deps "externals.coco-kbs.version")
+	image=$(get_from_kata_deps "externals.coco-kbs.image")
+	image_tag=$(get_from_kata_deps "externals.coco-kbs.image_tag")
+
+	if [ -d "$COCO_KBS_DIR" ]; then
+		rm -rf "$COCO_KBS_DIR"
+	fi
+
+	echo "::group::Clone the kbs sources"
+	git clone --depth 1 "${repo}" "$COCO_KBS_DIR"
+	pushd "$COCO_KBS_DIR"
+	git fetch --depth=1 origin "${version}"
+	git checkout FETCH_HEAD -b kbs_$$
+	echo "::endgroup::"
+
+	pushd kbs/config/kubernetes/
+
+	# Tests should fill kbs resources later, however, the deployment
+	# expects at least one secret served at install time.
+	echo "somesecret" > overlays/key.bin
+
+	echo "::group::Update the kbs container image"
+	install_kustomize
+	pushd base
+	kustomize edit set image "kbs-container-image=${image}:${image_tag}"
+	popd
+	echo "::endgroup::"
+
+	echo "::group::Deploy the KBS"
+	./deploy-kbs.sh
+	popd
+	popd
+
+	if ! kubectl -n "$ns" wait --for=condition=Ready --timeout=2m \
+		-l app=kbs pod; then
+		echo "::group::DEBUG - describe kbs pod"
+		kubectl -n "$ns" describe pod -l app=kbs || true
+		echo "::endgroup::"
+		return 1
+	fi
+
+	# Give some time for the service to start.
+	sleep 5
+	echo "::endgroup::"
+
+	# By default, the KBS service is reachable within the cluster only,
+	# thus the following healthy checker run from a pod.
+	echo "::group::Check the service healthy"
+	kbs_ip=$(kubectl get -o jsonpath='{.spec.clusterIP}' service/kbs -n "$ns" 2>/dev/null)
+	kbs_port=$(kubectl get -o jsonpath='{.spec.ports[0].port}' service/kbs -n "$ns" 2>/dev/null)
+	local pod=kbs-checker-$$
+	kubectl run "$pod" --image=busybox --restart=Never --wait=true -- \
+		sh -c "wget -O- \"${kbs_ip}:${kbs_port}\" || true"
```

Comment:
We prefer `curl` over `wget`.

Reply:
The busybox image doesn't come with `curl`